### PR TITLE
New flash interface review

### PR DESCRIPTION
**Pull request by LudwigKnuepfer**

### CODE REVIEW

**Comment by LudwigKnuepfer** [comment]
Maybe I missed something in the aftermath of the switch to mmap (useless casting), but it worked with/without Valgrind and with/without [existing] files.

File: cpu/native/periph/flash.c [hunk 1]
```diff
@@ -24,172 +27,76 @@
...
 #include "cpu-conf.h"
 #include "periph/flash.h"
 
 #include "native_internal.h"
 
-#define ENABLE_DEBUG (1)
+#define ENABLE_DEBUG (0)
 #include "debug.h"
 
-#define FLASH_ERROR_SIMULATION (1)
```
Comment (d00616) [inline cpu/native/periph/flash.c:41]: Later i plan implement an error simulation. I use this or an similar define to switch this feature on.

Comment (LudwigKnuepfer) [inline cpu/native/periph/flash.c:41]: I assumed as much, but I suggest to keep it out for now. It has no benefit unless there is an implementation using it.

Comment (LudwigKnuepfer) [inline cpu/native/periph/flash.c:41]: I don't insist on this commit though, I left it as a separate commit on purpose ;)

File: cpu/native/include/cpu-conf.h [hunk 2]
```diff
@@ -69,16 +69,14 @@ extern "C" {
...
-#define FLASH_NUM_PAGES			(256)	/**< Number of flash pages */
-#define FLASH_START_ADDRESS		(0x0)	/**< Starting address to calculate end of flash */
-#define FLASH_WRITE_ALIGN		(4)	/**< number of bytes must be written at once */
-#define FLASH_WRITES_PER_WORD		(2)	/**< how often a word can overwritten without flash erase */
-#define FLASH_ERASE_CYCLES		(20000) /**< Flash erase cycles descibed in datasheet */
-#define FLASH_ERASED_BIT_VALUE		(0)	/**< Value of erased bits 0|1 */
+#define FLASH_PAGE_SIZE         (1024)  /**< page size of flash memory */
+#define FLASH_NUM_PAGES         (256)   /**< number of flash pages */
+#define FLASH_WRITE_ALIGN       (4)     /**< number of bytes which must be written at once */
+#define FLASH_ERASED_BIT_VALUE  (0x00)  /**< value of erased bits */
```
Comment (d00616) [inline cpu/native/include/cpu-conf.h:78]: I plan to rename it to FLASH_ERASED_WORD_VALUE.

File: cpu/native/periph/flash.c [hunk 3]
```diff
@@ -198,102 +105,50 @@ uint8_t flash_memcpy(void *dest, const void *src, size_t n)
...
     if (page > FLASH_NUM_PAGES) {
         return FLASH_ERROR_ADDR_RANGE;
     }
 
-    // erase content
+    /* erase content */
     DEBUG("Erase page %d\n", page);
 
     for (ssize_t i = page * FLASH_PAGE_SIZE; i < ((page + 1) * FLASH_PAGE_SIZE); i++) {
 #if FLASH_ERASED_BIT_VALUE == 0
```
Comment (d00616) [inline cpu/native/periph/flash.c:142]: I remove ifdef here.

**Pull request merged**